src/sc/sidechain.cpp

```diff
@@ -713,17 +713,33 @@ int CSidechain::getNumBlocksForScFeeCheck()
 
 int CSidechain::getMaxSizeOfScFeesContainers()
 {
-    int epochLength = fixedParams.withdrawalEpochLength;
-    int numBlocks = getNumBlocksForScFeeCheck();
-
-    //assert(epochLength > 0);
-    if (epochLength == 0) {
-        maxSizeOfScFeesContainers = 1;
-    } else {
-        maxSizeOfScFeesContainers = numBlocks / epochLength;
-        if (maxSizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
-        {
-            maxSizeOfScFeesContainers++;
+    if (maxSizeOfScFeesContainers == -1) {
+        maxSizeOfScFeesContainers = getNumBlocksForScFeeCheck();
+
+        if (!isNonCeasing()) {
+            const int epochLength = fixedParams.withdrawalEpochLength;
+            assert(epochLength > 0);
+            if (epochLength == 0) {
+                maxSizeOfScFeesContainers = 1;
+            } else {
+                const int numBlocks = getNumBlocksForScFeeCheck();
+                maxSizeOfScFeesContainers = numBlocks / epochLength;
+                if (maxSizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
+                {
+                    maxSizeOfScFeesContainers++;
+                }
```

Comment:
Easy way to round up (ceil) integers dvisions (assuming operands are positive)

```suggestion
                const int numBlocks = getNumBlocksForScFeeCheck();
                maxSizeOfScFeesContainers = (numBlocks + epochLength - 1) / epochLength;
```
We save modulus operator to compute the remainder and an if branch
See https://godbolt.org/z/4nKq9sWbd

Reply:
This whole part has been updated and rewritten in 2522a6032b25bc34777cfd408f489058fb189cee